src/projects/trio/trio.cpp

```diff
@@ -0,0 +1,232 @@
+                continue;
+            auto label = mg.edges[eid]->getLabel();
+            if (haplotypes.find(label) != haplotypes.end()) {
+                if (haplotypes[label].haplotype == haplotype_) {
+                    if (mg.edges[eid]->isBridge() && mg.edges[eid]->size() < SAVED_BRIDGE_CUTOFF) {
+                        bridges ++;
+                        logger_.info() << "Skipping edge " << eid << " as bridge\n";
+                        continue;
+                    }
+                    removed_len += mg.edges[eid]->size();
+                    deleteEdgeHaplo(eid);
+
+                    logger_.trace() << "removing " << eid  << " label " << label << endl;
+                    removed ++;
+                    changed = true;
+                } else { 
+                    logger_.trace() << "skipping edge label" << label <<" "<< haplotypes[label].haplotype << endl;
+                }
+            } else {
+                logger_.trace() << "skipping edge NOT FOUND label" << label << endl;
+                
+            }
+        }
+    }
+    logger_.info() << "Saved " << bridges << "bridges\n";
+    logger_.info() << "Removed " << removed << " edges of haplo " << haplotype_  << " total len " << removed_len << endl;
+}
+
+void HaplotypeRemover::process() {
+    updateFixableHaplotypes();
+    updateAmbiguousHaplotypes();
+    removeHaplotype();
+    logger_.debug() << "removed \n";
+    mg.printEdgeGFA(out_dir / "before_clean.gfa", true);
+    cleanGraph();
+    mg.printEdgeGFA(out_dir / "mdbg.hpc.gfa", true);
+
+}
+
+
+std::experimental::filesystem::path simplifyHaplo(logging::Logger &logger, size_t threads,
+                                                  const std::experimental::filesystem::path &output_file,
+                                                  const std::experimental::filesystem::path &diplo_graph,
+                                                  const std::experimental::filesystem::path &haployak,
+                                                  const char haplotype,  const std::experimental::filesystem::path &corrected_reads,
+                                                  io::Library & reads,   const std::experimental::filesystem::path &dir) {
+    multigraph::MultiGraph mmg;
+    mmg.LoadGFA(diplo_graph, true);
+    multigraph::MultiGraph mg = mmg.DBG();
+    std::string out_name = "haplotype_";
+    out_name += other_haplo(haplotype);
+    std::experimental::filesystem::path out_dir = dir / out_name;
+    HaplotypeRemover hr(logger, mg, haployak, haplotype, out_dir);
+    hr.process();
+    mg.printEdgeGFA(output_file, true);
+
+//printing alignments and contigs, should be refactored
+    std::string out_aligns = out_name; out_aligns += ".alignments";
+    std::string out_contigs = out_name; out_contigs += ".fasta";
+    io::Library ref_lib;
+    multigraph::LJAPipeline pipeline (ref_lib);
+    size_t k = 5001;
+    std::vector<std::experimental::filesystem::path> uncompressed_results =
```

Comment:
Oh, wow. This is already one of the most time-consuming stages and you run it twice on full set of reads with half-graphs? The problem is that the output of your pipeline is not a graph. Can we fix that? Let's talk about it. Otherwise we can write another polishing stage that accepts set of contigs as an input and we can polish both paternal and maternal contigs together. But it still is important to have a graph until the very end.